examples/pytorch/tensor_parallel/README.md

```diff
@@ -0,0 +1,49 @@
+## Tensor Parallel and 2D Parallel
+
+This example shows how to apply tensor-parallelism to your model (here Llama 2 7B) with the `ModelParallelStrategy`, and how it can be combined with FSDP (2D parallelism).
+PyTorch 2.3+ and a machine with at least 4 GPUs and 24 GB memory each are required to run this example.
+
+```bash
+pip install 'torch>=2.3'
+```
+
+Navigate to this example folder and run the training script:
+
+```bash
+cd examples/pytorch/tensor_parallel
+python train.py
+```
+
+You should see an output like this:
+
+```
+GPU available: True (cuda), used: True
+TPU available: False, using: 0 TPU cores
+HPU available: False, using: 0 HPUs
+
+Number of model parameters: 6.7 B
+Starting training ...
+
+Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/4
+Initializing distributed: GLOBAL_RANK: 1, MEMBER: 2/4
+Initializing distributed: GLOBAL_RANK: 3, MEMBER: 4/4
+Initializing distributed: GLOBAL_RANK: 2, MEMBER: 3/4
+----------------------------------------------------------------------------------------------------
+distributed_backend=nccl
+All distributed processes registered. Starting with 4 processes
+----------------------------------------------------------------------------------------------------
+
+LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [4,5,6,7]
+LOCAL_RANK: 3 - CUDA_VISIBLE_DEVICES: [4,5,6,7]
+LOCAL_RANK: 1 - CUDA_VISIBLE_DEVICES: [4,5,6,7]
+LOCAL_RANK: 2 - CUDA_VISIBLE_DEVICES: [4,5,6,7]
```

Comment:
why is it using 4,5,6,7 here or is that just because you used those to run it?

Reply:
yes, because I share the machine and run on these GPUs. I modified it now to show 0,1,3,4 not to be confusing.